File: faktorial.c
#include <stdio.h>

long fac(int n)
{
    long hasil = 1;
    for (int i = 1; i <= n;i++){
         hasil*= i;
    }
       
    return hasil;
}
int main()
{
int n;
    printf("Masukkan bilangan bulat");
    scanf("%d",&n);
    printf("%d! = %ld\n",n,fac(n));

    return 0;
}

File: faktorial.dart
import 'dart:io';

int faktorial(int n) {
  int hasil = 1;
  for (int i = 1; i <= n; i++) {
    hasil *= i;
  }
  return hasil;
}

void main() {
  stdout.write("Masukkan bilangan bulat: ");
  int n = int.parse(stdin.readLineSync()!);

  print("Faktorial dari $n: ${faktorial(n)}");
}